--- salt-0.9.6/salt/grains/core.py_orig	2012-02-15 10:14:02.313348662 +0000
+++ salt-0.9.6/salt/grains/core.py	2012-02-15 10:15:23.770014906 +0000
@@ -355,6 +355,8 @@ def os_data():
                     grains['lsb_{0}'.format(match.groups()[0].lower())] = match.groups()[1].rstrip()
         if os.path.isfile('/etc/arch-release'):
             grains['os'] = 'Arch'
+        if os.path.isfile('/etc/pld-release'):
+            grains['os'] = 'PLD Linux'
         elif os.path.isfile('/etc/debian_version'):
             grains['os'] = 'Debian'
             if 'lsb_distrib_id' in grains:
